frontend/wallet/src/render/views/settings/CompressorSettings.re

```diff
@@ -0,0 +1,158 @@
+         <>
+           <div>
+             <div className=Styles.label> {React.string("Delegate to")} </div>
+             <TextField
+               label="Key"
+               value={
+                 Option.map(~f=PublicKey.toString, pubkey)
+                 |> Option.withDefault(~default="")
+               }
+               mono=true
+               onChange={value =>
+                 updateModal(_ =>
+                   Some(Delegate(Some(PublicKey.ofStringExn(value)), fee))
+                 )
+               }
+             />
+           </div>
+           <Spacer height=1. />
+           <div>
+             <div className=Styles.label>
+               {React.string("Transaction fee")}
+             </div>
+             <TextField.Currency
+               label="Fee"
+               value={
+                 Option.map(~f=Int64.to_string, fee)
+                 |> Option.withDefault(~default="")
+               }
+               placeholder="0"
+               onChange={value => {
+                 let serializedValue =
+                   switch (value) {
+                   | "" => None
+                   | nonEmpty => Some(Int64.of_string(nonEmpty))
+                   };
+                 updateModal(_ => Some(Delegate(pubkey, serializedValue)));
+               }}
+             />
+           </div>
+         </>
+       | Stake(fee) =>
+         <>
+           <TextField.Currency
+             label="Fee"
+             value={
+               Option.map(~f=Int64.to_string, fee)
+               |> Option.withDefault(~default="")
+             }
+             placeholder="0"
+             onChange={value => {
+               let serializedValue =
+                 switch (value) {
+                 | "" => None
+                 | nonEmpty => Some(Int64.of_string(nonEmpty))
+                 };
+               updateModal(_ => Some(Stake(serializedValue)));
+             }}
+           />
+           <Spacer height=1.5 />
+           <Alert
+             kind=`Warning
+             message="In order to recieve your staking reward, your computer must be on and running this wallet application."
+           />
+         </>
+       }}
+      <Spacer height=1.5 />
+      <div className=Css.(style([display(`flex)]))>
+        <Button
+          label="Cancel"
+          style=Button.Gray
+          onClick={_ => updateModal(_ => None)}
+        />
+        <Spacer width=1. />
+        <Button label="Save" style=Button.Green />
+      </div>
+    </Modal>;
+};
+
+[@react.component]
+let make = () => {
+  let (modalState, updateModal) = React.useState(() => None);
+  <div>
+    <h3 className=Theme.Text.Header.h3>
+      {React.string("Compression Settings")}
+    </h3>
+    <Spacer height=0.5 />
+    <Well>
+      <div className=Css.(style([display(`flex), alignItems(`flexEnd)]))>
+        <div className=Css.(style([flexGrow(1.)]))>
+          <div className=Styles.label> {React.string("Delegating to")} </div>
```

Comment:
hmm add todo to use modal state here to show "Staking" vs "Delegating to"?

Reply:
Yeah this will actually get set by the GraphQL response not modal state